snap_http/api.py

```diff
@@ -219,16 +219,68 @@ def revert_all(names: List[str]) -> SnapdResponse:
     return http.post("/snaps", {"action": "revert", "snaps": names})
 
 
-def remove(name: str) -> SnapdResponse:
+def remove(name: str,
+           purge: Optional[bool] = False,
+           terminate: Optional[bool] = False) -> SnapdResponse:
     """Uninstalls a snap identified by `name`."""
-    return http.post("/snaps/" + name, {"action": "remove"})
+    body = {
+        "action": "remove",
+        "purge": purge,
+        "terminate": terminate,
+    }
+
+    return http.post("/snaps/" + name, body)
 
 
 def remove_all(names: List[str]) -> SnapdResponse:
     """Uninstalls all snaps identified in `names`."""
     return http.post("/snaps", {"action": "remove", "snaps": names})
 
 
+def snapshots() -> SnapdResponse:
+    """Gets a list of all snapshots."""
+    return http.get("/snapshots")
+
+
+def forget_snapshot(id: str, snaps: Optional[List[str]] = None, users: Optional[List[str]] = None) -> SnapdResponse:
+    """Deletes a snapshot identified by `id`.
+
+    :param snap_id: The ID of the snapshot to delete.
+    """
+
+    body: Dict[str, Any] = {
+        "action": "forget",
+        "set": id
+    }
+
+    if snaps is not None:
+        body["snaps"] = snaps
+    if users is not None:
+        body["users"] = users
+
+    return http.post("/snapshots", body)
+
+
+def save_snapshot(
+    snaps: Optional[List[str]] = None,
+    users: Optional[List[str]] = None,
+) -> SnapdResponse:
+    """Saves a snapshot of the current state of the system.
+
+    :param name: The name of the snapshot.
+    :param users:  array of user names to whom snapshots are to be restricted .
+    :param snaps: Optional list of snaps to include in the snapshot.
+    """
+    body: Dict[str, Any] = {"action": "snapshot"}
+
+    if users:
```

Comment:
You explicitly check for `is not None` in the previous function, but here you're just checking truthiness. This means in the previous function the empty list `[]` would be included in `body`, but here it would not.

I'm unsure the implication of that from the snapd REST API perspective, but can we keep the behaviour consistent?